src/library/dao/trackdao.cpp

```diff
@@ -599,14 +599,14 @@ bool TrackDAO::addTracksAdd(TrackInfoObject* pTrack, bool unremove) {
                 }
             }
             // Regardless of whether we unremoved this track or not -- it's
-            // already in the library and so we need to skip it. Set the track's
-            // trackId so the caller can know it. TODO(XXX) this is a little
-            // weird because the track has whatever metadata the caller supplied
-            // and that metadata may differ from what is already in the
-            // database. I'm ignoring this corner case. rryan 10/2011
-            pTrack->setId(trackId);
```

Comment:
Moved to the end of this function to avoid uncontrolled setting of the track id.

Future: Setting the track id will only be allowed from within the TrackCache 
that needs to keep its internal maps synchronized.

On 02/01/2016 12:12 AM, Daniel Schürmann wrote:
>
> In src/library/dao/trackdao.cpp 
> <https://github.com/mixxxdj/mixxx/pull/885#discussion_r51369525>:
>
> > @@ -599,14 +599,14 @@ bool TrackDAO::addTracksAdd(TrackInfoObject* pTrack, bool unremove) {
> >                  }
> >              }
> >              // Regardless of whether we unremoved this track or not -- it's
> > -            // already in the library and so we need to skip it. Set the track's
> > -            // trackId so the caller can know it. TODO(XXX) this is a little
> > -            // weird because the track has whatever metadata the caller supplied
> > -            // and that metadata may differ from what is already in the
> > -            // database. I'm ignoring this corner case. rryan 10/2011
> > -            pTrack->setId(trackId);
>
> Where is this call gone?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/885/files#r51369525>.
>